go/vt/tabletserver/grpctabletconn/conn.go

```diff
@@ -139,50 +138,51 @@ func (conn *gRPCQueryClient) ExecuteBatch(ctx context.Context, target *querypb.T
 	return sqltypes.Proto3ToResults(ebr.Results), nil
 }
 
-type streamExecuteAdapter struct {
-	stream queryservicepb.Query_StreamExecuteClient
-	fields []*querypb.Field
-}
+// StreamExecute starts a streaming query to VTTablet.
+func (conn *gRPCQueryClient) StreamExecute(ctx context.Context, target *querypb.Target, query string, bindVars map[string]interface{}, options *querypb.ExecuteOptions, sendReply func(*sqltypes.Result) error) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
```

Comment:
The cancel is needed in the case where sendReply returns io.EOF or an error. If so, this function would return but the stream will remain open forever.